diff --git cmake/FindOpenCL.cmake cmake/FindOpenCL.cmake
index 57c4a94..917c5b0 100644
--- cmake/FindOpenCL.cmake
+++ cmake/FindOpenCL.cmake
@@ -32,11 +32,14 @@ ELSE (APPLE)
 
   IF (WIN32)
 
+    # [CMakeAll] OpenCL is distributed with CUDA
+    find_package(CUDA REQUIRED)
+
     FIND_PATH(OPENCL_INCLUDE_DIRS CL/cl.h )
     FIND_PATH(_OPENCL_CPP_INCLUDE_DIRS CL/cl.hpp )
 
     IF( ${OPENCL_INCLUDE_DIRS} STREQUAL "OPENCL_INCLUDE_DIRS-NOTFOUND" )
-     SET( SEARCH_PATH ${OPENCL_ROOT_DIR}/inc ${OPENCL_ROOT_DIR}/common/inc ${PATH} "C:/ProgramData/NVIDIA Corporation/NVIDIA GPU Computing SDK/OpenCL/common/inc" "$ENV{ATISTREAMSDKROOT}/include" "C:/Program Files (x86)/AMD APP/include")
+     SET( SEARCH_PATH ${CUDA_TOOLKIT_INCLUDE} ${OPENCL_ROOT_DIR}/inc ${OPENCL_ROOT_DIR}/common/inc ${PATH} "C:/ProgramData/NVIDIA Corporation/NVIDIA GPU Computing SDK/OpenCL/common/inc" "$ENV{ATISTREAMSDKROOT}/include" "C:/Program Files (x86)/AMD APP/include")
      FIND_PATH(OPENCL_INCLUDE_DIRS CL/cl.h PATHS ${SEARCH_PATH} )
      FIND_PATH(_OPENCL_CPP_INCLUDE_DIRS CL/cl.hpp PATHS ${SEARCH_PATH} )
     ENDIF( ${OPENCL_INCLUDE_DIRS} STREQUAL "OPENCL_INCLUDE_DIRS-NOTFOUND" )
diff --git code/CMakeLists.txt code/CMakeLists.txt
index 313e34f..cb96a9c 100644
--- code/CMakeLists.txt
+++ code/CMakeLists.txt
@@ -60,7 +60,19 @@ IF (CUDA_FOUND)
        rtkCudaFFTRampImageFilter.cu
        rtkCudaForwardProjectionImageFilter.cu
        rtkCudaBackProjectionImageFilter.cu )
+  # [CMakeAll] reduce number of include directories on Windows
+  if(WIN32)
+    get_directory_property(SAVED_INCLUDE_DIRS INCLUDE_DIRECTORIES)
+    set(CUSTOM_INCLUDE_DIRS ${SAVED_INCLUDE_DIRS})
+    list(REMOVE_ITEM CUSTOM_INCLUDE_DIRS ${ITK_INCLUDE_DIRS})
+    list(APPEND CUSTOM_INCLUDE_DIRS ${ITKKWSys_INCLUDE_DIRS})
+    list(APPEND CUSTOM_INCLUDE_DIRS ${ITKCommon_INCLUDE_DIRS})
+    set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "${CUSTOM_INCLUDE_DIRS}")
+  endif()
   CUDA_COMPILE (rtkcuda_CUDA_WRAPPERS ${rtkcuda_CUDA_FILES})
+  if(WIN32)
+    set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "${SAVED_INCLUDE_DIRS}")
+  endif()
   ADD_LIBRARY(rtkcuda
               rtkCudaFDKBackProjectionImageFilter.cxx
               rtkCudaFFTRampImageFilter.cxx
diff --git code/rtkCudaFFTRampImageFilter.cu code/rtkCudaFFTRampImageFilter.cu
index 21f4ee8..9bcc897 100644
--- code/rtkCudaFFTRampImageFilter.cu
+++ code/rtkCudaFFTRampImageFilter.cu
@@ -90,8 +90,8 @@ CUDA_fft_convolution(const int3 &inputDimension, float *projection, cufftComplex
 
   // Thread Block Dimensions
   int tBlock_x = 16;
-  int tBlock_y = 8;
-  int tBlock_z = 8;
+  int tBlock_y = 4;
+  int tBlock_z = 4;
 
   int  blocksInX = (fftDimension.x - 1) / tBlock_x + 1;
   int  blocksInY = (fftDimension.y - 1) / tBlock_y + 1;
diff --git code/rtkHndImageIO.cxx code/rtkHndImageIO.cxx
index 8d59982..c4bcde2 100644
--- code/rtkHndImageIO.cxx
+++ code/rtkHndImageIO.cxx
@@ -18,6 +18,13 @@
 
 // std include
 #include <stdio.h>
+// [CMakeAll] define missing types on Windows
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32)
+  typedef __int32 int32_t;
+  typedef unsigned __int32 uint32_t;
+  typedef __int64 int64_t;
+  typedef unsigned __int64 uint64_t;
+#endif
 
 #include "rtkHndImageIO.h"
 #include <itkMetaDataObject.h>
